18F/identity-design-system
# Replace Gulp with NPM scripts

## 🛠 Summary of changes

Replaces [Gulp](https://gulpjs.com/) as the build task runner with equivalent NPM scripts.

**Why?**

- Gulp hasn't received a release since 2019. Similarly, its plugin ecosystem is not consistently maintained, which has caused us some difficulty in the past.
- Currently we split build tasks across `Makefile`, `package.json`, and `gulpfile.js`. This consolidates to eliminate one of those.
- A good chunk of our `gulpfile.js` tasks were targeted at supporting the public documentation site, with broad browser compatibility. Since we don't need that any longer, this helps simplify. (Removes Autoprefixer, Uglify, etc.)
- Major reduction in dependency tree depth (see lines removed)
   - Helps maintainability by eliminating a common source of security advisories
      - This pull request alone mitigates from "11 vulnerabilities (5 moderate, 6 high)" to "2 moderate severity vulnerabilities"
- Migrates to use [`@18f/identity-build-sass`](https://github.com/18F/identity-idp/tree/main/app/javascript/packages/build-sass) to further dog-food this project and ensure consistency across projects

## 📜 Testing Plan

* Verify that the Cloud.gov Pages preview continues to work
   * Notably, verify both stylesheets and JavaScript appear to be functional (e.g. Accordions collapse)
* Verify there are no regressions in the built package output
   * `make build` should produce `build/esm` and `build/cjs` containing ESM and CJS JavaScript respectively
